pilot/pkg/config/cloudfoundry/controller.go

```diff
@@ -279,21 +283,24 @@ func createVirtualService(gatewayNames []string, route *copilotapi.RouteWithBack
 	}
 }
 
-func createHTTPRoute(route *copilotapi.RouteWithBackends) *networking.HTTPRoute {
-	return &networking.HTTPRoute{
-		Route: []*networking.DestinationWeight{
-			{
-				Destination: &networking.Destination{
-					Host: route.GetHostname(),
-					Port: &networking.PortSelector{
-						Port: &networking.PortSelector_Number{
-							Number: 8080,
-						},
-					},
+func createDestinationWeight(route *copilotapi.RouteWithBackends) *networking.DestinationWeight {
+	return &networking.DestinationWeight{
+		Destination: &networking.Destination{
+			Host:   route.GetHostname(),
+			Subset: route.GetCapiProcessGuid(),
```

Comment:
where is the destination rule for this?

Reply:
Hello rshriram,

We did not change anything in Destination Rules, we basically updated Virtual Services to contain all destinations in the same route array. Could you please clarify what are you looking for?

Thank you.

Reply:
If you specify subsets in the destination weight, you need a destinationRule that defines the subset. I don't see a destination rule in this code. I see just the virtualService. Put another way, if you define this config, you will see 503s. Is the code to generate destination rules, already present? If so, please point me to it

Reply:
The destinationRule and subsets are defined here. https://github.com/istio/istio/blob/bd17e46e3ad4151c41427aa37c8717c55b1bf360/pilot/pkg/config/cloudfoundry/controller.go#L222-L227. The destination rule and virtual services are linked through hostnames.